xonsh/main.py

```diff
@@ -251,9 +251,11 @@ def _pprint_displayhook(value):
         builtins._ = value
         return
     env = builtins.__xonsh__.env
+    printed_val = None
     if env.get("PRETTY_PRINT_RESULTS"):
         printed_val = pretty(value)
-    else:
+    if not isinstance(printed_val, str):
```

Comment:
Should be `if printed_val is None` for speed

Reply:
But that's the point - when this issue happens, `pretty(value)` returns `value.xonsh_display()` which is itself a mock object.
So the way to make sure pretty succeeded is to check if it returned a `str` like it should.

Reply:
Oh I see.